api/types/network/network.go

```diff
@@ -28,6 +28,14 @@ type EndpointIPAMConfig struct {
 	LinkLocalIPs []string `json:",omitempty"`
 }
 
+// Copy makes a copy of the endpoint ipam config
+func (cfg *EndpointIPAMConfig) Copy() *EndpointIPAMConfig {
```

Comment:
WDYT about having `pkg/deepcopy` that implements generic deepcopy via `reflect`? (in another PR)